DogTales/dog/player/player_controller.cpp

```diff
@@ -0,0 +1,45 @@
+#include <dog/player/player_controller.hpp>
+#include <algorithm>
+#include <cassert>
+
+namespace dog {
+
+PlayerController::PlayerController(bave::NotNull<bave::App const*> app) : m_app(app) {}
+
+void PlayerController::bind_throttle(std::string_view const id, Throttle const throttle) {
+	if (throttle.hi == bave::Key::eUnknown) { return; }
+	auto it = m_mappings.find(id);
+	if (it == m_mappings.end()) {
+		auto [i, _] = m_mappings.insert_or_assign(std::string{id}, std::vector<Mapping>{});
```

Comment:
could you explain the syntax here? what is `[i, _]` ?

Reply:
That's a structured binding. You can use it to "destructure" an aggregate type or a `std::tuple`:

```cpp
struct Foo {
  int x{};
  char y{};
};

auto f = Foo{42, 'x'};
auto [i, c] = f;
// i == 42, c == 'x'
```

The underscore is just convention for "I don't care about this member", from the language/compiler's point of view it's just an identifier (name of a variable).